src/main/java/com/google/sps/servlets/ScheduledInterviewServlet.java

```diff
@@ -83,36 +96,81 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   // Send the request's contents to Datastore in the form of a new ScheduledInterview object.
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String requestedInterviewerId = request.getParameter("interviewer");
-    String requestedIntervieweeId = request.getParameter("interviewee");
-    String userEmail = UserServiceFactory.getUserService().getCurrentUser().getEmail();
-    String userId = UserServiceFactory.getUserService().getCurrentUser().getUserId();
-    // Since UserId does not have a valid Mock, if the id is null (as when testing), it will be
-    // replaced with this hashcode.
-    if (userId == null) {
-      userId = String.format("%d", userEmail.hashCode());
+    String intervieweeEmail = userService.getCurrentUser().getEmail();
+    String intervieweeId = getUserId();
+
+    InterviewPostRequest postRequest;
+    try {
+      postRequest = new Gson().fromJson(getJsonString(request), InterviewPostRequest.class);
+    } catch (Exception JsonSyntaxException) {
+      response.sendError(400);
+      return;
     }
-    // The default key for a scheduledInterview being stored in datastore
-    long defaultKey = -1;
-    if ((!requestedInterviewerId.equals(userId)) && (!requestedIntervieweeId.equals(userId))) {
-      response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+    if (!postRequest.allFieldsPopulated()) {
+      response.sendError(400);
       return;
     }
+
+    String interviewerCompany = postRequest.getCompany();
+    String interviewerJob = postRequest.getJob();
+    String utcStartTime = postRequest.getUtcStartTime();
+
+    TimeRange range;
+
     try {
-      ScheduledInterview scheduledInterview =
-          ScheduledInterview.create(
-              defaultKey,
-              new TimeRange(
-                  Instant.parse(request.getParameter("startTime")),
-                  Instant.parse(request.getParameter("endTime"))),
-              requestedInterviewerId,
-              requestedIntervieweeId);
-      scheduledInterviewDao.create(scheduledInterview);
-      return;
+      range =
+          new TimeRange(
+              Instant.parse(utcStartTime), Instant.parse(utcStartTime).plus(1, ChronoUnit.HOURS));
     } catch (DateTimeParseException e) {
       response.sendError(400, e.getMessage());
       return;
     }
+
+    List<Availability> availabilitiesInRange =
+        availabilityDao.getInRangeForAll(range.start(), range.end());
+    List<Person> allAvailableInterviewers =
+        interviewerServlet.getPossiblePeople(availabilitiesInRange, range);
```

Comment:
Calling a method on an instance of another servlet is kind of weird, as is having one servlet own an instance of another servlet. If possible, I think you should make getPossiblePeople a static method and call `ShowInterviewersServlet.getPossiblePeople()`

Reply:
Done. Involved passing around some daos, but it works.